app/Services/AdventuresLeagueAdapter.php

```diff
@@ -153,12 +127,43 @@ private function getEntries($data, $characterId)
                     'description' => (array_key_exists(6, $data[$i])) ? $data[$i][6] : "",
                     'author_id' => Auth::id(),
                     'tier' => 0,
-                    'counted' => 0,
                 ];
-                $item = new Item($itemData);
-                $item->save();
+                Item::create($itemData);
             }
         }
         return $entries;
     }
+
+    private function getEntryData($characterId, $data, $type)
+    {
+        $entryData = [
+            'user_id' => Auth::id(),
+            'adventure_id' => null,
+            'campaign_id' => null,
+            'character_id' => $characterId,
+            'event_id' => null,
+            'dungeon_master_id' => null,
+            'levels' => 0,
+            ];
+
+        if (is_null($data)) {
+            $defaultEntryData = [
+                'date_played' => now(),
+                'gp' => 0,
+                'downtime' => 0,
+                'type' => Entry::TYPE_GAME,
+            ];
+            return array_merge($entryData, $defaultEntryData);
+        } else {
+            $populatedEntryData = [
+                'dungeon_master' => (array_key_exists(12, $data)) ? (float)$data[12] : "",
+                'location' => (array_key_exists(11, $data)) ? (float)$data[11] : "",
+                'date_played' => ($data[3] == "") ? now() : $data[3],
+                'gp' => (array_key_exists(7, $data)) ? (float)$data[7] : 0,
+                'downtime' => (array_key_exists(8, $data)) ? (float)$data[8] : 0,
+                'type' => $type,
+            ];
```

Comment:
might be more readable to wrap `$data` in a collection and then use the [`get()` method](https://laravel.com/docs/8.x/collections#method-get), but i'd say thats mostly a matter of opinion
```suggestion
            $data = collect($data)
            $populatedEntryData = [
                'dungeon_master' => (float) $data->get(12),
                'location' => (float) $this->get(11),
                'date_played' => Carbon::parse($data->get(3)),
                'gp' => (float) $data->get(7, 0),
                'downtime' => (float) $data->get(8, 0),
                'type' => $type,
            ];
```

Reply:
@m-triassi Can you tell me what happens if the index does not exist for a given collection through the get method? I couldn't find documentation online.

Reply:
@pascwhale haha you didn't read closely enough, first line of the get methods docs:
> The get method returns the item at a given key. **If the key does not exist, null is returned**

Reply:
I like this implementation, its a lot cleaner. I added it. Thanks!